### PR TITLE
Disable -Werror by default in nGraph build

### PR DESCRIPTION
**Pull request by dcaballe**
This PR disables treating warnings as errors by default so that
nGraph can be built with compiler versions that are not currently
tested in CI. We could enable -Werror in CI for those compiler versions
we want to target for warning-free builds.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -143,7 +143,7 @@ option(NGRAPH_PLAIDML_STATIC_LIB_ENABLE "Enable build PlaidML backend as a stati
 option(NGRAPH_DYNAMIC_COMPONENTS_ENABLE "Enable dynamic loading of components" TRUE)
 option(NGRAPH_NATIVE_ARCH_ENABLE "Enable build for native archtecture" TRUE)
 option(NGRAPH_EXPORT_TARGETS_ENABLE "Enable exporting nGraph cmake export targets" TRUE)
-option(NGRAPH_WARNINGS_AS_ERRORS "Make all nGraph compile-time warnings into errors" TRUE)
+option(NGRAPH_WARNINGS_AS_ERRORS "Make all nGraph compile-time warnings into errors" FALSE)
```
Comment (rkimballn1) [inline CMakeLists.txt]: Does CI set warnings as errors for at least some of the runs? If not we need to get that enabled before we merge this.

**Comment by rkimballn1** [comment]
I added Blocked label because I want to make sure that CI is setting WARNINGS_AS_ERRORS for at least some of the runs

**Comment by rkimballn1** [comment]
I just checked the log files and warnings as errors is enabled for at least the one run I look at so we are all good. Thanks for fixing this.

**Comment by aslepko** [comment]
TESTNOW

**Pull request merged**